staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -335,8 +335,8 @@ func (d *DiscoveryClient) ServerVersion() (*version.Info, error) {
 func (d *DiscoveryClient) OpenAPISchema() (*openapi_v2.Document, error) {
 	data, err := d.restClient.Get().AbsPath("/openapi/v2").SetHeader("Accept", mimePb).Do().Raw()
 	if err != nil {
-		if errors.IsForbidden(err) || errors.IsNotFound(err) {
-			// single endpoint not found/registered in old server, try to fetch old endpoint
+		if errors.IsForbidden(err) || errors.IsNotFound(err) || errors.IsNotAcceptable(err) {
```

Comment:
@MaciekPytel 

As the comment said:

> // TODO(roycaihw): remove this in 1.11

We are not going to support old apiserver in 1.11.
So this is only necessary for 1.10.

Reply:
Sorry, but I don't consider TODO a guarantee that it will actually happen in 1.11. I don't want to fix this in 1.10 only to break it again in 1.11 if plans change. The fix applies just as well to current state of master and the change can be cherry-picked from master without any additional effort.

Please create a PR to master and follow a normal process (https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md) to cherry-pick it.

Reply:
> Please create a PR to master and follow a normal process 

I think this PR should not goes into the master...
In the master branch, we should delete the `if{}` scope.

Reply:
In the 1.11 branch we do not support old apiserver  any more, so we delete the `if{}` scope.

Reply:
I am OK to close this pull request, because I think users should use the same version of kubectl and kube-apiserver.

Reply:
/close